---
title: "RNA_protein_association"
author: "Howard Cen"
date: "2023-10-30"
output: html_document
---


```{r packages}
#install.packages("tidyverse")
library(tidyverse)

#install.packages("readxl")
library(readxl)

#install.packages("writexl")
library(writexl)

#install.packages("openxlsx")
library(openxlsx)


library(ggrepel) # for label text in ggplot2

#install.packages("GGally")
#library("GGally") # For the ggpair() to plot pair-wise correlation panels

#BiocManager::install("EnhancedVolcano")
#library(EnhancedVolcano)


library(scales) # for log10 breaks (and scale to z score for heatmap)

library(ggpubr) # for stat_cor


#BiocManager::install("org.Hs.eg.db")
library(org.Hs.eg.db)

#BiocManager::install("clusterProfiler")
#BiocManager::install("ReactomePA")
library("clusterProfiler")
library("ReactomePA")


#install.packages("Hmisc")
library(Hmisc) # for correlation rcorr()


# For marginal density plot
#install.packages("ggExtra")
library(ggExtra)
library(RColorBrewer)
library(cowplot)


#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install("DESeq2")
#library("DESeq2")

#install.packages("msigdbr")
library(msigdbr) # load MSigDB gene sets, v7.5.1 (released January 2022)

# cran repository
#install.packages("cocor") #package to calculate if 2 correlations are different # Diedenhofen, B. & Musch, J. (2015). cocor: A Comprehensive Solution for the Statistical Comparison of Correlations. PLoS ONE, 10(4): e0121945. doi:10.1371/journal.pone.0121945
# alternative repository
#install.packages("cocor",repo="http://comparingcorrelations.org/repo")
require(cocor) # package for testing if correlations are different

setwd(dirname(rstudioapi::getActiveDocumentContext()$path)) ##Set working directory to where this file is.
getwd()


```

```{r lm adjust batch function}

lm_pr_batch <- function(pro.dat,rna.dat, batch.dat){
  
  # create empty data frames to save results
  lm_result <- data.frame(matrix(nrow = 0, ncol = 0))
  # run linear regression for each gene-trait pair, using diabetes status and batch as covariate
  for (i in 1:ncol(pro.dat)){
    data <- cbind(pro.dat[i], rna.dat[i], batch.dat["batch"])
    colnames(data) <- c("protein","RNA","batch")
    model <- lm(protein ~ RNA + batch, data=data)
    coefficients.df <- as.data.frame(summary(model)[["coefficients"]])
    
    lm_result[i,1] <- coefficients.df["RNA","Estimate"]
    lm_result[i,2] <- coefficients.df["RNA","Pr(>|t|)"]
    lm_result[i,3] <- summary(model)[["adj.r.squared"]]
    lm_result[i,4] <- summary(model)$fstatistic %>% {unname(pf(.[1],.[2],.[3],lower.tail=F))}
  }
  
  colnames(lm_result) <- c("coeff_rna", "p_rna", "R2_adj", "p_model")
  rownames(lm_result) <- colnames(rna.dat)
  
  lm_result$padj_rna <- p.adjust(lm_result$p_rna, method = 'fdr')
  lm_result$padj_model <- p.adjust(lm_result$p_model, method = 'fdr')
  
  lm_result <- lm_result %>%
  mutate(p_sig = ifelse(p_rna >= 0.05, "ns",
                      ifelse(coeff_rna >0, "positive", "negative")),
         padj_sig = ifelse(padj_rna >= 0.05, "ns",
                         ifelse(coeff_rna > 0, "positive", "negative")) )
  
  lm_result$ENSG <- rownames(lm_result)
  lm_result$entrez <- mapIds(org.Hs.eg.db, keys=lm_result$ENSG, column="ENTREZID", keytype="ENSEMBL", multiVals="first")
  lm_result$symbol <- mapIds(org.Hs.eg.db, keys=lm_result$ENSG, column="SYMBOL", keytype="ENSEMBL", multiVals="first")
  lm_result$fullname <- mapIds(org.Hs.eg.db, keys=lm_result$ENSG, column="GENENAME", keytype="ENSEMBL", multiVals="first")

  write.csv(lm_result, "output/LOG2protein_VSTrna_lm_results_batch.csv")
}

  
```



```{r calculate RNA protein lm batch}
pro.cor.within <- read.csv("output/lm_cor_input_LOG2protein50.csv", row.names = 1)
rna.cor.within <- read.csv("output/lm_cor_input_VSTrna50.csv", row.names = 1)

View(pro.cor.within)
all(colnames(pro.cor.within)==colnames(rna.cor.within))
#traits.log <- read.csv(file = "output/islets_traits_log.csv", check.names = F)
batch.df <- read.delim("input/Proteomics_RNAseqSamples_Batches.txt", header = F)
colnames(batch.df) <- c("donor", "batch")
View(batch.df)

batch.cor <- batch.df[match(rownames(rna.cor.within), batch.df$donor),]
View(batch.cor)
all(batch.cor$donor==rownames(rna.cor.within))


# function to calculate linear regression for every gene and trait, using gene as dependent, trait as predictor, and diabetes status as a categorical variable
#gene.df <- pro.log
#trait.df <- traits.pro.cor
#trait.adjust <- traits.pro

#gene.df <- rna.cor.order
#trait.df <- traits.rna.cor
#trait.adjust <- traits.rna
pro.dat <- pro.cor.within
rna.dat <- rna.cor.within
batch.dat <- batch.cor
i=1
lm_pr_batch(pro.dat = pro.cor.within, rna.dat = rna.cor.within, batch.dat = batch.cor)

lm_result <- read.csv("output/LOG2protein_VSTrna_lm_results_batch.csv", row.names = 1)

View(lm_result)

table(lm_result$p_sig)

# lm results:
# positive negative       ns 
#    2227       83     5278 

# correlation results   
#positive negative       ns 
#2273      123     5192 

df <- table(lm_result$p_sig) %>% as.data.frame()
View(df)
round(df$Freq/sum(df$Freq)*100,2)

# lm results
# 29.35  1.09 69.56

# correlation results  
# 29.96  1.62 68.42

table(lm_result$padj_sig)

# lm results  
# positive negative       ns 
#    1416       19     6153

# correlation results      
#positive negative       ns 
#1422       37     6129 
    
df <- table(lm_result$padj_sig) %>% as.data.frame()
View(df)



round(df$Freq/sum(df$Freq)*100,2)

# lm results  
# 18.66  0.25 81.09

# correlation results  
#18.74  0.49 80.77

```

```{r ORA function}

ora_human <- function(x, all_genes, name){ 
  
  library(msigdbr) # load MSigDB gene sets, v7.5.1 (released January 2022)
  library("clusterProfiler")
  library("ReactomePA")
  #install.packages("openxlsx")
  library(openxlsx)
  
  #KEGG ORA
  KEGG <- enrichKEGG(gene         = x$entrez,
                     organism     = 'hsa',
                     universe      = all_genes,
                     pAdjustMethod="BH", 
                     pvalueCutoff=1, 
                     qvalueCutoff=1,
                     minGSSize = 15,
                     maxGSSize = 500)
  KEGG <- setReadable(KEGG, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  KEGG.df <- as.data.frame(KEGG)
  head(KEGG.df)

  #Reactome ORA
  react <- enrichPathway(gene         = x$entrez,
                         organism     = 'human',
                         universe      = all_genes,
                         minGSSize = 15,
                         maxGSSize = 500,
                         pvalueCutoff=1, pAdjustMethod="BH", 
                         qvalueCutoff=1)
  react <- setReadable(react, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  react.df <- as.data.frame(react)
  head(react.df)
  #react.df2 <- react.df
  #react.df2$Description <- gsub(x = react.df2$Description, pattern = "\\ ",replacement = "_") 
  #react.df2$Description <- gsub(x = react.df2$Description, pattern = "\\,",replacement = ".")
  #write.xlsx2(react.df, file="output/correlated_genes_ORA.xlsx", sheetName = "Reactome",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gobp <- enrichGO(gene        = x$entrez,
                    universe      = all_genes,
                    OrgDb         = org.Hs.eg.db,
                    ont           = "BP",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 1,
                    qvalueCutoff  = 1,
                    minGSSize = 15,
                    maxGSSize = 500,
                    readable      = TRUE)
  gobp.df <- as.data.frame(gobp)
  head(gobp.df)
  #write.xlsx2(gobp.df, file="output/correlated_genes_ORA.xlsx", sheetName = "GO_BP",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gomf <- enrichGO(gene       = x$entrez,
                    universe      = all_genes,
                    OrgDb         = org.Hs.eg.db,
                    ont           = "MF",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 1,
                    qvalueCutoff  = 1,
                    minGSSize = 15,
                    maxGSSize = 500,
                    readable      = TRUE)
  gomf.df <- as.data.frame(gomf)
  head(gomf.df)
  #write.xlsx2(gomf.df, file="output/correlated_genes_ORA.xlsx", sheetName = "GO_MF",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gocc <- enrichGO(gene      = x$entrez,
                    universe      = all_genes,
                    OrgDb         = org.Hs.eg.db,
                    ont           = "CC",
                    pAdjustMethod = "BH",
                    pvalueCutoff  = 1,
                    qvalueCutoff  = 1,
                    minGSSize = 15,
                    maxGSSize = 500,
                    readable      = TRUE)
  gocc.df <- as.data.frame(gocc)
  head(gocc.df)
  #write.xlsx2(gocc.df, file="output/correlated_genes_ORA.xlsx", sheetName = "GO_CC",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  # MSigDb ORA
  h_t2g.h <- msigdbr(species = "Homo sapiens", category = "H") %>% 
    dplyr::select(gs_name, entrez_gene)
  head(h_t2g.h)
  
  
  hallmark <- enricher(x$entrez,
                       TERM2GENE=h_t2g.h,
                        universe  = all_genes,
                       pvalueCutoff = 1,
                       qvalueCutoff = 1,
                       minGSSize = 15,
                       maxGSSize = 500
                       )

  hallmark <- setReadable(hallmark, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  hallmark.df <- as.data.frame(hallmark)
  View(hallmark.df)
  #
  h_t2g.c8 <- msigdbr(species = "Homo sapiens", category = "C8") %>% 
    dplyr::select(gs_name, entrez_gene)
  head(h_t2g.c8)
  
  c8 <- enricher(x$entrez,
                 TERM2GENE=h_t2g.c8,
                 universe      = all_genes,
                 pvalueCutoff = 1,
                 qvalueCutoff = 1,
                 minGSSize = 15,
                 maxGSSize = 500)
  c8 <- setReadable(c8, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  c8.df <- as.data.frame(c8)

wb <- createWorkbook()

# Add the data frames to separate sheets
addWorksheet(wb, "MSigDB_hallmark")
writeData(wb, "MSigDB_hallmark", hallmark.df)

addWorksheet(wb, "KEGG")
writeData(wb, "KEGG", KEGG.df)

addWorksheet(wb, "Reactome")
writeData(wb, "Reactome", react.df)

addWorksheet(wb, "GO_BP")
writeData(wb, "GO_BP", gobp.df)

addWorksheet(wb, "GO_MF")
writeData(wb, "GO_MF", gomf.df)

addWorksheet(wb, "GO_CC")
writeData(wb, "GO_CC", gocc.df)

addWorksheet(wb, "MSigDB_C8_celltype")
writeData(wb, "MSigDB_C8_celltype", c8.df)


# Save the workbook to a file
saveWorkbook(wb, paste0("output/ORA_", name, "_LOG2pro50_VSTrna50.xlsx"), overwrite = TRUE)

cat(paste0("ORA results saved in - output/ORA_", name, "_LOG2pro50_VSTrna50.xlsx"))
}

```

```{r lm results ORA}
lm_result <- read.csv("output/LOG2protein_VSTrna_lm_results_batch.csv", row.names = 1)

ora_human(lm_result[lm_result$padj_sig=="positive", ], all_genes = as.character(lm_result$entrez), name = "lm_positive")

ora_human(lm_result[lm_result$padj_sig=="ns", ], all_genes = as.character(lm_result$entrez), name = "lm_ns")

ora_human(lm_result[lm_result$padj_sig=="negative", ], all_genes = as.character(lm_result$entrez), name = "lm_negative")

```


```{r plot lm within gene}
lm_result <- read.csv("output/LOG2protein_VSTrna_lm_results_batch.csv", row.names = 1)

lm_result$p_sig <- factor(lm_result$p_sig,levels = c("positive","negative","ns"))
lm_result$padj_sig <- factor(lm_result$padj_sig,levels = c("positive","negative","ns"))

p <- ggplot(lm_result,
            aes(y=-log10(padj_rna), 
                
                x=R,
                colour= padj_sig
                
            )
) +  
  geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.8) +
  labs(x="Coefficients", 
       y=expression(-log[10]~(adj~p.value))
  ) +
  scale_colour_manual(values=c("red","blue","grey")) +
#  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
#                     limits= c(-0.5,1)) +
  #  scale_y_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Coefficients" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  #coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = FALSE, groupFill = FALSE,
                 margins ="x", colour="#E69F00",fill=alpha("#E69F00",0.5)) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2


{pdf(file = "figures/vstRNA_LOG2protein_linearRegression_batch.pdf",
    width = 6.5, 
    height = 6.5)

p2
dev.off()}
```

```{r calculate within gene correlation}
#
pro.filter <- read.table(file="output/protein_filter50.txt",sep = "\t",header = T)
rna.norm <- read.table(file="output/Norm_vst_islets_oneBatch50.txt",sep = "\t",
                        header = T,row.names = 1)

View(pro.filter)
View(rna.norm)
dim(pro.filter)
dim(rna.norm)

common.donor <- intersect(colnames(pro.filter),colnames(rna.norm)) # 90 donors
length(common.donor)
common.gene <- intersect(pro.filter$ENSG,rownames(rna.norm))
length(common.gene) # 7609

dim(rna.norm) #90 x 20806 (5 counts 50% samples)

View(pro.filter)

pro.cor.ENSG <- pro.filter[,-c(1,3)] %>%
  filter(ENSG %in% common.gene) %>%
  select(all_of(c("Protein.Ids","Genes","ENSG",common.donor)))
colnames(pro.cor.ENSG)

# removed dup protein ENSG, kept the one with highest abundance
pro.cor.ENSG <- pro.cor.ENSG[rowSums(is.na(pro.cor.ENSG[,-(1:3)]))<(0.5*ncol(pro.cor.ENSG[,-(1:3)])),]
pro.cor.ENSG <- pro.cor.ENSG[order(rowSums(pro.cor.ENSG[,-(1:3)]),decreasing = T),]# order row sum from high to low abundance
dim(pro.cor.ENSG) # 7599 proteins
pro.cor.ENSG <- pro.cor.ENSG[!duplicated(pro.cor.ENSG$ENSG),] # remove duplicated ENSG by keeping the most abundant one
dim(pro.cor.ENSG) # 7588 proteins after removing duplicated ENSG

rownames(pro.cor.ENSG) <- pro.cor.ENSG$ENSG
dim(pro.cor.ENSG) #7588 proteins detected in >=50% of the 90 common donors. 

# log2 transform protein
pro.cor.within <- as.data.frame(log2(t(pro.cor.ENSG[,-(1:3)])))

# (not used) without log transform protein - worse correlation
#pro.cor.within <- as.data.frame((t(pro.cor.ENSG[,-(1:3)])))

rna.cor.within <- as.data.frame(t(rna.norm[pro.cor.ENSG$ENSG,common.donor]))
dim(rna.cor.within) #7588 rna
View(pro.cor.within)
View(rna.cor.within)

identical(colnames(pro.cor.within),colnames(rna.cor.within))
identical(rownames(pro.cor.within), rownames(rna.cor.within))

write.csv(pro.cor.within, "output/lm_cor_input_LOG2protein50.csv")
write.csv(rna.cor.within, "output/lm_cor_input_VSTrna50.csv")

pro.cor.within <- read.csv("output/lm_cor_input_LOG2protein50.csv", row.names = 1)
rna.cor.within <- read.csv("output/lm_cor_input_VSTrna50.csv", row.names = 1)

# calculated the correlation

R <- 0
p.value <- 0
index <- 1

for (i in 1:ncol(rna.cor.within)){
  res.cor <- rcorr(as.matrix(cbind(pro.cor.within[,i],rna.cor.within[,i])),type="pearson")
  R[index] <- res.cor$r[1,2]
  p.value[index] <- res.cor$P[1,2]
  index <- index+1
}
pro.rna.withinGene <- data.frame(ENSG=colnames(rna.cor.within),R,p.value)
head(pro.rna.withinGene)
dim(pro.rna.withinGene) # 7588 genes

pro.rna.withinGene$p.adj <- p.adjust(pro.rna.withinGene$p.value, method = 'fdr')

pro.rna.withinGene <- pro.rna.withinGene %>%
  mutate(p_sig=ifelse(p.value>=0.05,"ns",
                      ifelse(R>0,"positive","negative")),
         padj_sig=ifelse(p.adj>=0.05,"ns",
                         ifelse(R>0,"positive","negative"))
  )
pro.rna.withinGene$p_sig <- factor(pro.rna.withinGene$p_sig,levels = c("positive","negative","ns"))
pro.rna.withinGene$padj_sig <- factor(pro.rna.withinGene$padj_sig,levels = c("positive","negative","ns"))

View(pro.cor.ENSG)
identical(pro.cor.ENSG$ENSG,pro.rna.withinGene$ENSG)
pro.rna.withinGene$uniprot <- pro.cor.ENSG$Protein.Ids
pro.rna.withinGene$entrez <- mapIds(org.Hs.eg.db, keys=pro.rna.withinGene$ENSG, column="ENTREZID", keytype="ENSEMBL", multiVals="first")
pro.rna.withinGene$symbol <- mapIds(org.Hs.eg.db, keys=pro.rna.withinGene$ENSG, column="SYMBOL", keytype="ENSEMBL", multiVals="first")
pro.rna.withinGene$fullname <- mapIds(org.Hs.eg.db, keys=pro.rna.withinGene$ENSG, column="GENENAME", keytype="ENSEMBL", multiVals="first")

pro.rna.withinGene <- pro.rna.withinGene[order(pro.rna.withinGene$R,decreasing = T),]
View(pro.rna.withinGene)
dim(pro.rna.withinGene) # 7588 rows
length(unique(pro.rna.withinGene$ENSG)) #7588 genes
#table(unique(pro.rna.withinGene$ENSG) %in% rownames(raw.counts.islets.oneBatch.5)) #all 6918 genes reliably detected by RNAseq
# TRUE 
# 6990

table(pro.rna.withinGene$p_sig)
#positive negative       ns 
#2273      123     5192 
df <- table(pro.rna.withinGene$p_sig) %>% as.data.frame()
View(df)
round(df$Freq/sum(df$Freq)*100,2)
# 29.96  1.62 68.42

table(pro.rna.withinGene$padj_sig)
#positive negative       ns 
#1422       37     6129 
df <- table(pro.rna.withinGene$padj_sig) %>% as.data.frame()
View(df)
round(df$Freq/sum(df$Freq)*100,2)
#18.74  0.49 80.77

write.table(pro.cor.within,file = "output/LOGpro50_withinGene_df.txt",
            sep = "\t",row.names = F)
write.table(rna.cor.within,file = "output/vstRNA50_withinGene_df.txt",
            sep = "\t",row.names = F)

write.table(pro.rna.withinGene,file = "output/LOGpro50_vstRNA50_withinGene_correlation.txt",
            sep = "\t",row.names = F)


pro.rna.withinGene <- read.table(file = "output/LOGpro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)


write.table(pro.rna.withinGene,file = "output/LOG2pro50_vstRNA50_withinGene_correlation.txt",
            sep = "\t",row.names = F)


pro.rna.withinGene <- read.table(file = "output/LOG2pro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)
View(pro.rna.withinGene)




###########
Positive=pro.rna.withinGene[pro.rna.withinGene$padj_sig=="positive",] %>%
  select(all_of(c("uniprot","ENSG","entrez","symbol","fullname")))
Negative=pro.rna.withinGene[pro.rna.withinGene$padj_sig=="negative",] %>%
  select(all_of(c("uniprot","ENSG","entrez","symbol","fullname")))
Background=pro.rna.withinGene %>%
  select(all_of(c("uniprot","ENSG","entrez","symbol","fullname")))
View(Positive)
View(Negative)
View(Background)
write.csv(Positive, file="output/islet_proteins_positive.csv", row.names = F)
write.csv(Negative, file="output/islet_proteins_negative.csv", row.names = F)
write.csv(Background, file="output/islet_proteins_background.csv", row.names = F)
Positive <- read.csv(file="output/islet_proteins_positive.csv")
Negative <- read.csv(file="output/islet_proteins_negative.csv")
Background <- read.csv(file="output/islet_proteins_background.csv")
View(Positive)
```


```{r visualize/validate within gene abundance}
pro.cor.within <- read.table(file = "output/LOGpro50_withinGene_df.txt",
                                 sep = "\t",header = T)
rna.cor.within <- read.table(file = "output/vstRNA50_withinGene_df.txt",
                                 sep = "\t",header = T)

ggplot(data=data.frame(x=rna.cor.within$ENSG00000168237, 
                       y=pro.cor.within$ENSG00000168237
                       ),
       aes(y=y, x=x) ) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(
    #aes(colour=R),
    #color="grey20",
    size=0.8,
    alpha= 0.4 
  ) +
  geom_smooth(
    #se=FALSE,
    method=lm
  ) +
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
  )+
  
  labs(x= "RNA level (VST)", 
       y= "Protein abundance (log10)"
  ) +
  scale_colour_manual(values=c("red","blue","grey60")) +
  #scale_fill_gradient2(high = "red3",mid = "white",low = "blue3",midpoint = 0,
  #  space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "colour")+
  
  # scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
  #                labels = trans_format("log10", math_format(10^.x))) +
  #  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
  #                labels = trans_format("log10", math_format(10^.x))) +
  #  annotation_logticks(sides = "bl") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
#    limits= c(0,20)) +
#scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
#scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+

theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position=
      "left",
    #"none",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))

```


```{r plot within gene correlation}

pro.rna.withinGene <- read.table(file = "output/LOGpro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)
pro.rna.withinGene$p_sig <- factor(pro.rna.withinGene$p_sig,levels = c("positive","negative","ns"))
pro.rna.withinGene$padj_sig <- factor(pro.rna.withinGene$padj_sig,levels = c("positive","negative","ns"))

p <- ggplot(pro.rna.withinGene,
            aes(y=-log10(p.adj), 
                
                x=R,
                colour= padj_sig
                
            )
) +  
  geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.8) +
  labs(x="R (protein vs RNA of each gene)", 
       y=expression(-log[10]~(adj~p.value))
  ) +
  scale_colour_manual(values=c("red","blue","grey")) +
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  #  scale_y_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  #coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = FALSE, groupFill = FALSE,
                 margins ="x", colour="#E69F00",fill=alpha("#E69F00",0.5)) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2

png(file = "figures/vstRNA_LOGprotein_correlation_each_gene_padj_oneBatch.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)

{pdf(file = "figures/vstRNA_LOG2protein_correlation_each_gene_padj_oneBatch.pdf",
    width = 6.5, 
    height = 6.5)

p2
dev.off()}

```


```{r GSEA calculate}
pro.rna.withinGene <- read.table(file = "output/LOGpro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)
View(pro.rna.withinGene)
# option1 rank by R value
#genelist <- pro.rna.withinGene$R
# (used this) option2 rank by signed -log10 p value
genelist <- sign(pro.rna.withinGene$R) * (-log10(pro.rna.withinGene$p.value))
# option3 rank by R*(-log10 p value)
#genelist <- (pro.rna.withinGene$R) * (-log10(pro.rna.withinGene$p.value))


# entrez id as names of the gene list
names(genelist) <- pro.rna.cor.each.gene.oneBatch$entrez
genelist = sort(genelist, decreasing = TRUE)
head(genelist)
genelist[genelist==Inf] <- 20

# Run GSEA for each comparison

x <- genelist

gsea <- function(x){
  
  gse.kegg <- gseKEGG(
    geneList=x,
    organism = "hsa",
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    keyType = "kegg"
  )
  gse.kegg <<- setReadable(gse.kegg, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gse.kegg.df <<- as.data.frame(gse.kegg)
  View(gse.kegg.df)
  
  gseReactome <- gsePathway(
    geneList=x,
    organism = "human",
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    verbose = TRUE)
  gseReactome <<- setReadable(gseReactome, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gseReactome.df <<- as.data.frame(gseReactome)
  head(gseReactome.df)
  #write.xlsx2(gseReactome.df, file="output/GSEA_.xlsx", sheetName = "Reactome",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  
  gseGO.bp <- gseGO(
    geneList=x,
    ont = "BP",
    OrgDb = org.Hs.eg.db,
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    verbose = TRUE,
    keyType = "ENTREZID"
  )
  gseGO.bp <<- setReadable(gseGO.bp, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gseGO.bp.df <<- as.data.frame(gseGO.bp)
  head(gseGO.bp.df)
  #write.xlsx2(gseGO.bp.df, file="output/GSEA_.xlsx", sheetName = "GO_BP",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gseGO.mf <- gseGO(
    geneList=x,
    ont = "MF",
    OrgDb = org.Hs.eg.db,
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    verbose = TRUE,
    keyType = "ENTREZID"
  )
  gseGO.mf <<- setReadable(gseGO.mf, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gseGO.mf.df <<- as.data.frame(gseGO.mf)
  head(gseGO.mf.df)
  #write.xlsx2(gseGO.mf.df, file="output/GSEA_.xlsx", sheetName = "GO_MF",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  
  gseGO.cc <- gseGO(
    geneList=x,
    ont = "CC",
    OrgDb = org.Hs.eg.db,
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    verbose = TRUE,
    keyType = "ENTREZID"
  )
  gseGO.cc <<- setReadable(gseGO.cc, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gseGO.cc.df <<- as.data.frame(gseGO.cc)
  head(gseGO.cc.df)
  #write.xlsx2(gseGO.cc.df, file="output/GSEA_.xlsx", sheetName = "GO_CC",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gse.mkegg <- gseMKEGG(
    geneList=x,
    organism = "hsa",
    minGSSize = 15, 
    maxGSSize = 500,
    pvalueCutoff = 1,
    keyType = "kegg"
  )
  gse.mkegg <<- setReadable(gse.mkegg, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gse.mkegg.df <<- as.data.frame(gse.mkegg)
  head(gse.mkegg)
  #write.xlsx2(gse.mkegg.df, file="output/GSEA_.xlsx", sheetName = "mKEGG",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  gse.wp <- gseWP(x, organism = "Homo sapiens",pvalueCutoff = 1)
  gse.wp <<- setReadable(gse.wp, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  gse.wp.df <<- as.data.frame(gse.wp)
  head(gse.wp)
  #write.xlsx2(gse.wp.df, file="output/GSEA_.xlsx", sheetName = "mKEGG",
  #            col.names = TRUE, row.names = TRUE, append = TRUE)
  
  
  # mSigDB Hallmark gene sets  
  #msigdbr_show_species()
  #m_df <- msigdbr(species = "Mus musculus")
  #head(m_df, 2) %>% as.data.frame
  
  # H: hallmark gene sets
  # C1: positional gene sets
  # C2: curated gene sets
  # C3: motif gene sets
  # C4: computational gene sets
  # C5: GO gene sets
  # C6: oncogenic signatures
  # C7: immunologic signatures
  
  # MSigDb GSEA
  
  #msigdbr_show_species()
  h_t2g.h <- msigdbr(species = "Homo sapiens", category = "H") %>% 
    dplyr::select(gs_name, entrez_gene)
  head(h_t2g.h)
  
  gse.hallmark <- GSEA(x, #geneList 
                       TERM2GENE = h_t2g.h)
  
  gse.hallmark <<- setReadable(gse.hallmark, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  
  gse.hallmark.df <- as.data.frame(gse.hallmark)
  
  View(gse.hallmark.df)
  #

  h_t2g.c8 <- msigdbr(species = "Homo sapiens", category = "C8") %>% 
    dplyr::select(gs_name, entrez_gene)
  head(h_t2g.c8)
  
  gse.c8 <- GSEA(x, TERM2GENE = h_t2g.c8)
  
  gse.c8 <<- setReadable(gse.c8, OrgDb = org.Hs.eg.db, keyType="ENTREZID") # The geneID column is translated from EntrezID to symbol
  
  gse.c8.df <- as.data.frame(gse.c8)
  
  View(gse.c8.df)
  
}

## 

wb <- createWorkbook()

# Add the data frames to separate sheets
addWorksheet(wb, "MSigDB_hallmark")
writeData(wb, "MSigDB_hallmark", gse.hallmark.df)

addWorksheet(wb, "KEGG")
writeData(wb, "KEGG", gse.kegg.df)

addWorksheet(wb, "Reactome")
writeData(wb, "Reactome", gseReactome.df)

addWorksheet(wb, "GO_BP")
writeData(wb, "GO_BP", gseGO.bp.df)

addWorksheet(wb, "GO_MF")
writeData(wb, "GO_MF", gseGO.mf.df)

addWorksheet(wb, "GO_CC")
writeData(wb, "GO_CC", gseGO.cc.df)

addWorksheet(wb, "KEGG_module")
writeData(wb, "KEGG_module", gse.mkegg.df)

addWorksheet(wb, "Wiki")
writeData(wb, "Wiki", gse.wp.df)

addWorksheet(wb, "MSigDB_C8_celltype")
writeData(wb, "MSigDB_C8_celltype", gse.c8.df)


# Save the workbook to a file
saveWorkbook(wb, "output/GESA_p_LOGpro50_VSTrna50.xlsx", overwrite = TRUE)

```

```{r RNA protein correlation ORA calculation}

## ORA ----------------
pro.rna.withinGene <- read.table(file = "output/LOG2pro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)

pro.rna.withinGene <- read.table(file = "output/LOGpro50_vstRNA50_withinGene_correlation.txt",
                                 sep = "\t",header = T)

View(pro.rna.withinGene)
#geneList <- pro.rna.withinGene$log2FoldChange
#names(geneList) <- as.character(pro.rna.withinGene$entrez)
#geneList = sort(geneList, decreasing = TRUE)
#head(geneList)


ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="positive",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor_positive")

ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="ns",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor_ns")

ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="negative",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor_negative")

ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="positive",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor10_positive")

ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="negative",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor10_negative")

ora_human(pro.rna.withinGene[pro.rna.withinGene$padj_sig=="ns",], all_genes = as.character(pro.rna.withinGene$entrez), name = "cor10_ns")

```


```{r plot ORA results from within gene correlation}

#ora.gocc.pos <- read_excel("output/ORA_positive_LOGpro50_VSTrna50.xlsx", sheet = 2)
#ora.gocc.ns <- read_excel("output/ORA_ns_LOGpro50_VSTrna50.xlsx", sheet = 2)
#ora.gocc.neg <- read_excel("output/ORA_negative_LOGpro50_VSTrna50.xlsx", sheet = 2)

ora.gocc.pos <- read_excel("output/ORA_cor_positive_LOG2pro50_VSTrna50.xlsx", sheet = 6)
ora.gocc.ns <- read_excel("output/ORA_cor_ns_LOG2pro50_VSTrna50.xlsx", sheet = 6)

View(ora.gocc.pos)
View(ora.gocc.ns)


# GO_CC positive
df <-  ora.gocc.pos %>%
  filter(qvalue<0.001) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "Positive"

# GO_CC non-significant
df <-  ora.gocc.ns %>%
  filter(qvalue<0.001) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "NS"



# (common)
df$GeneRatio <- unname(sapply(df$GeneRatio, function(x) eval(parse(text = x)))) 


View(df)

# plot positive
p.pos <- ggplot(df, 
                         aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = GeneRatio, color = logFDR)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 0, 
    mid= "white",
    high = "red", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(size = "Gene Ratio", color="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black')) +
  guides(size = guide_legend(order=1))

p.pos

# plot ns
p.ns <- ggplot(df, 
                         aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = GeneRatio, color = logFDR)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 0, 
    mid= "white",
    high = "grey20", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(size = "Gene Ratio", color="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black')) +
  guides(size = guide_legend(order=1))

p.ns



#install.packages("cowplot")
library(cowplot)
#plot_grid(p.pos, p.ns, p.neg, ncol = 1, align = "v",rel_heights=c(7.5,4.5,3))

plot_grid(p.pos, p.ns, ncol = 1, align = "v",rel_heights=c(7.5,4.5))

ggsave(filename="figures/vstRNA_LOG2protein_within_gene_cor_ORA.pdf",width=19,height=28,units="cm")


```

```{r plot ORA results from within gene lm}

#ora.gocc.pos <- read_excel("output/ORA_positive_LOGpro50_VSTrna50.xlsx", sheet = 2)
#ora.gocc.ns <- read_excel("output/ORA_ns_LOGpro50_VSTrna50.xlsx", sheet = 2)
#ora.gocc.neg <- read_excel("output/ORA_negative_LOGpro50_VSTrna50.xlsx", sheet = 2)

ora.gocc.pos <- read_excel("output/ORA_lm_positive_LOG2pro50_VSTrna50.xlsx", sheet = 6)
ora.gocc.ns <- read_excel("output/ORA_lm_ns_LOG2pro50_VSTrna50.xlsx", sheet = 6)

View(ora.gocc.pos)
View(ora.gocc.ns)


# GO_CC positive
df <-  ora.gocc.pos %>%
  filter(qvalue<0.0001) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "Positive"

# GO_CC non-significant
df <-  ora.gocc.ns %>%
  filter(qvalue<0.0001) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "NS"



# (common)
df$GeneRatio <- unname(sapply(df$GeneRatio, function(x) eval(parse(text = x)))) 


View(df)

# plot positive
p.pos <- ggplot(df, 
                         aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = GeneRatio, color = logFDR)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 0, 
    mid= "white",
    high = "red", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(size = "Gene Ratio", color="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black')) +
  guides(size = guide_legend(order=1))

p.pos

# plot ns
p.ns <- ggplot(df, 
                         aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = GeneRatio, color = logFDR)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 0, 
    mid= "white",
    high = "grey20", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(size = "Gene Ratio", color="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black')) +
  guides(size = guide_legend(order=1))

p.ns



#install.packages("cowplot")
library(cowplot)
#plot_grid(p.pos, p.ns, p.neg, ncol = 1, align = "v",rel_heights=c(7.5,4.5,3))

plot_grid(p.pos, p.ns, ncol = 1, align = "v",rel_heights=c(7.5,4.5))

ggsave(filename="figures/ORA_vstRNA_LOG2protein_lm_within_gene.pdf",width=21,height=29,units="cm")


```


```{r (not used) plot GSEA results from within gene correlation}

GSEA.cor.each.gene.oneBatch.hallmark <- read_excel("input/GSEA_p_oneBatch.xlsx", sheet = 1)
GSEA.cor.each.gene.oneBatch.gocc <- read_excel("input/GSEA_p_oneBatch.xlsx", sheet = 6)


# Hallmark
df <-  GSEA.cor.each.gene.oneBatch.hallmark %>%
  filter(qvalue<0.005) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "mSigDB_Hallmark"



#(common scipts)
df$Description <- gsub(x =df$Description, pattern = "HALLMARK_",replacement = "") 
df$Description <- gsub(x =df$Description, pattern = "_",replacement = " ") 
df$Description <- str_to_sentence(df$Description)

View(df)

# plot hallmark
p.hallmark <- ggplot(df, 
                         aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = logFDR,
                 colour = NES)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 1, 
    low = "blue",
    mid= "white",
    high = "red", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(color = "NES",size="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black'))

p.hallmark

# GO CC
df <-  GSEA.cor.each.gene.oneBatch.gocc %>%
  filter(qvalue<0.005) %>% 
  mutate(logFDR=log10(qvalue)*(-1))
df$library <- "GO_CC"

#plot GO CC
p.gocc <- ggplot(df, 
                     aes(x=1, y = fct_reorder(Description, -pvalue))) + 
  geom_point(aes(size = logFDR,
                 colour = NES)) +
  theme_bw(base_size = 16) +
  scale_color_gradient2(#limits=c(0,5), 
    midpoint = 1, 
    low = "blue",
    mid= "white",
    high = "red", 
    space = "Lab" )+
  ylab(NULL)+
  xlab(NULL)+
  theme(axis.ticks.x = element_blank(),
        axis.text.y = element_text(colour = "black",size=14),
        #axis.text.x = element_text(colour = "black",size=14),
        axis.text.x = element_blank(),
        
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        
        legend.title = element_text(color = "black", size = 13))+
  labs(color = "NES",size="-log10(FDR)") +
  theme(panel.grid.minor.y = element_blank())+
  theme(panel.grid.minor.x  = element_blank()) +
  facet_grid(. ~ library) +
  theme(strip.background =element_rect(fill=alpha(cbPalette[1], 0.2) ))+
  theme(strip.text = element_text(colour = 'black'))

p.gocc

#install.packages("cowplot")
library(cowplot)
plot_grid(p.hallmark, p.gocc, ncol = 1, align = "v",rel_heights=4:5)

ggsave(filename="figures/RNA_protein_within_gene_GSEA_Hallmark_GOCC_oneBatch.png",width=17.5,height=18,units="cm",dpi=400)
ggsave(filename="figures/RNA_protein_within_gene_GSEA_Hallmark_GOCC_oneBatch.pdf",width=17.5,height=18,units="cm")


```



# Across gene correlations

```{r process}
# plot protein RNA correlation by TPM and length (one batch) =============

#write.table(tpms, file = "output/RNAseq_TPM.txt", sep = "\t", row.names = F) 

tpms <- read.table(file = "output/RNAseq_TPM.txt", sep = "\t", header = T)
dim(tpms)
View(tpms)

rna.norm <- read.table(file="output/Norm_vst_islets_oneBatch50.txt",sep = "\t",
                        header = T,row.names = 1)
View(rna.norm)

tpms.filter <- tpms %>%
  filter(ensembl_gene_id %in% rownames(rna.norm))

write.table(tpms.filter, file = "output/RNAseq_TPM_filter50.txt", sep = "\t", row.names = F) 

tpm.oneBatch <- tpms.filter[,c(1:8,colnames(tpms.filter) %in% colnames(rna.norm))]
View(tpm.oneBatch)
write.table(tpm.oneBatch, file = "output/RNAseq_TPM_filter50_oneBatch.txt", sep = "\t", row.names = F) 


tpm.filter
dim(tpms.5)
colnames(tpms)

tpms.oneBatch <- tpms %>% 
  filter(ensembl_gene_id %in% colnames(rna.cor2)) %>%
  select(c(colnames(tpms)[1:8],rownames(rna.cor2))) 
tpms.oneBatch$mean.TPM <- rowMeans(tpms.oneBatch[,-(1:8)])
tpms.oneBatch <- tpms.oneBatch %>%
  relocate(mean.TPM, .after=GeneName)
dim(tpms.oneBatch) # 6917x99 #one gene was probably not annotated at the TPM calculation
colnames(rna.cor2)[!colnames(rna.cor2) %in% tpms.oneBatch$ensembl_gene_id] #ENSG00000112096 no longer in Ensembl database. SOD2 gene.
View(tpms.oneBatch)

write.table(tpms.oneBatch, file = "output/RNA.islets.TPM.oneBatch.txt",
      sep = "\t", row.names = F) 

tpm.oneBatch <- read.table(file = "output/RNA.islets.TPM.oneBatch.txt",
            sep = "\t", header = T)
View(tpm.oneBatch)
dim(tpm.oneBatch) # 6917 99
tpms.islets.oneBatch <- tpms %>%
  select(c(colnames(tpms)[1:8],rownames(rna.cor2))) %>%
  filter(ensembl_gene_id %in% tpms.5$ensembl_gene_id)
dim(tpms.islets.oneBatch)
colnames(tpms.islets.oneBatch) 
write.table(tpms.islets.oneBatch, file = "output/RNA_islets_TPM_oneBatch_all.txt",
            sep = "\t", row.names = F) 
write.table(tpms.islets.oneBatch, file = "output/RNA_islets_TPM_oneBatch_5.txt",
            sep = "\t", row.names = F)



tpm.islet.oneBatch <- read.table(file = "output/RNA_islets_TPM_oneBatch_5.txt",
            sep = "\t", header = T)

View(pro.rna.cor.each.gene.oneBatch)

pro.rna.cor.each.gene.oneBatch <- pro.rna.cor.each.gene.oneBatch %>%
  left_join(tpms.oneBatch[,c(1:9)], by=c("ENSG"="ensembl_gene_id"))
View(pro.rna.cor.each.gene.oneBatch)
```

```{r}
# plot TPM vs R with separately colored  positive and negative groups--------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=mean.TPM, 
              x=R,
              
              colour= padj_sig 
                     #p_sig
            )
) +  
  geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.8) +
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Mean TPM"
  ) +
  scale_colour_manual(values=c("red","blue","grey")) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="top",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  #coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = TRUE, groupFill = TRUE,
                 margins ="y"#, colour="#E69F00",fill=alpha("#E69F00",0.5)
) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2

png(file = "figures/RNA_protein_correlation_each_gene_padj_TPM_oneBatch.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)
png(file = "figures/RNA_protein_correlation_each_gene_punadj_TPM_oneBatch.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)

p2

dev.off()

# plot TPM vs R with continuously colored  R ----------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=mean.TPM, 
              x=R #,colour= R
            )
) +
  geom_point(aes(colour=R),size=0.8, alpha= 0.8) +
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Mean TPM"
  ) +
  #scale_colour_manual(values=c("red","blue","grey")) +
  scale_fill_gradient2(
    high = "red3",
    mid = "white",
    low = "blue3",
    midpoint = 0,
    space = "Lab",
    na.value = "grey50",
    guide = "colourbar",
    aesthetics = "colour"
  )+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
    
    geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
    #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
    
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="top",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) 
  #coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
 # guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))

p

png(file = "figures/RNA_protein_correlation_each_gene_TPM_oneBatch_gradient.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)

p

dev.off()
#ggsave(filename="figures/RNA_protein_correlation_each_gene_punadj_TPM.png",width=6.5,height=5,units="in",dpi=600)

# plot correlation of transcript length vs R ------------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=mean.TPM, 
              x=R,
              
              # colour= p_sig
            )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4 , color="grey20"
  ) +
  geom_smooth(se=TRUE,
              method=lm
  ) +
  
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
  )+
  
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Mean TPM") +
  scale_colour_manual(values=c("red","blue","grey60")) +
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p


ggsave(filename="figures/RNA_protein_correlation_each_gene_TPM_oneBatch_pearson.png",width=5,height=5,units="in",dpi=600)

# plot transcript length vs R with separately colored  positive and negative groups -----------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=transcript_length, 
              x=R,
              
              colour=# padj_sig 
                      p_sig
            )
) +  
  geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.8) +
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Transcript length (bp)") +
  scale_colour_manual(values=c("red","blue","grey")) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="top",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  #coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = TRUE, groupFill = TRUE,
                 margins ="y"#, colour="#E69F00",fill=alpha("#E69F00",0.5)
) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2

png(file = "figures/RNA_protein_correlation_each_gene_padj_length_oneBatch.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)
png(file = "figures/RNA_protein_correlation_each_gene_punadj_length_oneBatch.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)

p2

dev.off()

# plot transcript length vs R with continuously colored  R -----------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=transcript_length, 
              x=R #,colour= R
            )
) +
  geom_point(aes(colour=R),size=0.8, alpha= 0.8) +
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Transcript length (bp)") +
  #scale_colour_manual(values=c("red","blue","grey")) +
  scale_fill_gradient2(
    high = "red3",
    mid = "white",
    low = "blue3",
    midpoint = 0,
    space = "Lab",
    na.value = "grey50",
    guide = "colourbar",
    aesthetics = "colour"
  )+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="top",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) 
#coord_axes_inside(labels_inside = TRUE) +    #https://stackoverflow.com/questions/17753101/center-x-and-y-axis-with-ggplot2
# guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))

p

png(file = "figures/RNA_protein_correlation_each_gene_length_oneBatch_gradient.png",
    width = 6.5, 
    height = 6.5, 
    units = "in", res = 600)

p

dev.off()


# plot correlation of transcript length vs R --------------------

p <- ggplot(pro.rna.cor.each.gene.oneBatch,
            aes(
              y=transcript_length, 
              x=R,
              
              # colour= p_sig
            )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4 , color="grey20"
  ) +
  geom_smooth(se=TRUE,
              method=lm
              ) +
 
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
           )+
  
  labs(x= "R (protein vs RNA of each gene)", 
       y= "Transcript length (bp)") +
  scale_colour_manual(values=c("red","blue","grey60")) +
  scale_x_continuous(breaks=c(-0.4,-0.2,0,0.2,0.4,0.6,0.8,1),
                     limits= c(-0.5,1)) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p


ggsave(filename="figures/RNA_protein_correlation_each_gene_padj_length_oneBatch_pearson.png",width=5,height=5,units="in",dpi=600)



```

```{r calculate mean abundance, CoV}
# global RNA protein correlation of all genes =====================

View(pro.rna.cor.each.gene.oneBatch)
View(pro.all.df)
View(pro.islet.filter)
View(uniprot.size)
colnames(pro.islet.filter)
colnames(uniprot.size)


pro.filter <- read.table(file="output/protein_filter50.txt",sep = "\t",header = T)
View(pro.filter)
str(pro.filter)

# Downloaded uniprot data from here: Programmatic access - Retrieving entries via queries: https://www.uniprot.org/help/api_queries
uniprot <- read_excel("input/uniprot-compressed_true_download_true_fields_accession_2Creviewed_2C-2022.11.28-17.35.59.13.xlsx")
View(uniprot)
uniprot.size <- uniprot %>%
  dplyr::select(Entry,`Entry Name`,`Protein names`, `Gene Names`, Length, Mass) %>%
  #separate(`Entry Name`, into = c("Gene",NA)) %>%
  separate(`Gene Names`,into = c("Gene1","Gene2","Gene3","Gene4","Gene5",
                                 "Gene6","Gene7","Gene8","Gene9","Gene10",
                                 "Gene11","Gene12","Gene13","Gene14","Gene15",
                                 "Gene16","Gene17","Gene18","Gene19","Gene20")) %>%
  pivot_longer(cols = Gene1:Gene20, names_to = "name", values_to = "Genes") %>%
  filter(!is.na(Genes)) %>%
  dplyr::select(-name)
View(uniprot.size)
dup <- uniprot.size %>% group_by(Entry) %>% filter(n()>1) 
View(dup)
# each uniprot ID might have different gene names, by everything else are the same (length...)
uniprot.size.uni <- uniprot.size[!duplicated(uniprot.size$Entry),]
View(uniprot.size.uni)

write.csv(uniprot.size, "input/uniprot_protein_size.csv", row.names = F)

#
pro.filter.norm <- pro.filter %>%
  left_join(uniprot.size.uni[,-c(2,6)], by=c("Protein.Ids"="Entry")) 
pro.filter.norm[which(pro.filter.norm$Genes=="INS"), "Length"] <- 51

pro.filter.norm <- pro.filter.norm %>%
  mutate(mean.signal=rowMeans(pro.filter[,-c(1:4)], na.rm = T)) %>%
  mutate(mean.signal.length = mean.signal/Length) %>%
  relocate(`Protein names`:mean.signal.length,.after=ENSG)

View(pro.filter.norm)
str(pro.filter.norm)

pro.filter.norm <- pro.filter.norm %>% 
  mutate_at(vars(10:143),list(norm.length = ~./Length)) # every sample & gene is divided by length

pro.filter.norm$protein.CoV <- apply(pro.filter.norm[,10:143], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
pro.filter.norm$protein.SD <- apply(pro.filter.norm[,10:143], 1, function(x) sd(x, na.rm=T))

pro.filter.norm$protein.length.CoV <- apply(pro.filter.norm[,144:277], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
pro.filter.norm$protein.length.SD <- apply(pro.filter.norm[,144:277], 1, function(x) sd(x, na.rm=T))

pro.filter.norm <- pro.filter.norm %>%
  relocate(protein.CoV:protein.length.SD, .after = mean.signal.length)

write.csv(pro.filter.norm, "output/proteomics50_signal_CoV_anno.csv", row.names = F)


#
#write.table(tpms, file = "output/RNAseq_TPM.txt", sep = "\t", row.names = F)
tpms <- read.table(file = "output/RNAseq_TPM.txt", sep = "\t", header = T)
View(tpms)
rna.norm <- read.table(file="output/Norm_vst_islets_oneBatch50.txt",sep = "\t", header = T,row.names = 1)

tpms.filter <- tpms %>% filter(ensembl_gene_id %in% rownames(rna.norm))
View(tpms.filter)
write.table(tpms.filter, file = "output/RNAseq_TPM_filter50.txt", sep = "\t", row.names = F)

tpm.oneBatch <- tpms.filter[,c(colnames(tpms.filter)[1:8], intersect(colnames(tpms.filter), colnames(rna.norm)))]
View(tpm.oneBatch)
write.table(tpm.oneBatch, file = "output/RNAseq_TPM_filter50_oneBatch.txt", sep = "\t", row.names = F)

tpm.oneBatch.norm <- tpm.oneBatch%>%
  mutate(mean.TPM = rowMeans(tpm.oneBatch[,-c(1:8)], na.rm = T)) %>%
  relocate(mean.TPM,.after = GeneName)

tpm.oneBatch.norm$RNA.CoV <- apply(tpm.oneBatch.norm[,10:99], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
tpm.oneBatch.norm$RNA.SD <- apply(tpm.oneBatch.norm[,10:99], 1, function(x) sd(x, na.rm=T))

tpm.oneBatch.norm <- tpm.oneBatch.norm %>%
  relocate(RNA.CoV:RNA.SD, .after = mean.TPM)
View(tpm.oneBatch.norm)

write.csv(tpm.oneBatch.norm, "output/RNAseq50_TPM_CoV_anno.csv", row.names = F)

#
dup <- pro.filter.norm %>% group_by(ENSG) %>% filter(n()>1) # 13 ENSG duplicated, kept the most abundant one

pro.filter.norm[order( rowSums(pro.filter.norm[,-(1:13)], na.rm = T), decreasing = T ), ]
pro.filter.norm1 <- pro.filter.norm[!duplicated(pro.filter.norm$ENSG), ]

tpm.oneBatch.norm[order( rowSums(tpm.oneBatch.norm[,-(1:11)], na.rm = T), decreasing = T ), ]
tpm.oneBatch.norm1 <- tpm.oneBatch.norm[!duplicated(tpm.oneBatch.norm$ensembl_gene_id), ]

pro.rna.mean.cov <- inner_join(pro.filter.norm1[1:13], tpm.oneBatch.norm1[1:11], by = c("ENSG"="ensembl_gene_id") )
colnames(pro.rna.mean.cov)

pro.rna.mean.cov <- pro.rna.mean.cov %>%
  select(c("Protein.Ids", "Genes", "ENSG", "Protein names", "Length", "Mass", 
           "mean.signal", "mean.signal.length", "protein.CoV", "protein.SD", "protein.length.CoV", "protein.length.SD",
           "external_gene_name", "description", "gene_biotype", "entrezgene_id", 
           "transcript_length", "mean.TPM", "RNA.CoV", "RNA.SD"))
dim(pro.rna.mean.cov) # 7608
write.csv(pro.rna.mean.cov, "output/proteomics50_RNAseq50_anno_mean_CoV.csv", row.names = F)
```

```{r plot across gene correlation}
pro.rna.mean.cov <- read.csv("output/proteomics50_RNAseq50_anno_mean_CoV.csv")
pro.rna.mean.cov <- pro.rna.mean.cov %>% 
  mutate(xmin=mean.TPM,
         xmax=mean.TPM+RNA.SD,
         ymin=mean.signal.length,
         ymax=mean.signal.length+protein.length.SD
  )
# plot CoV as error bar (didn't use this)
pro.rna.mean.cov <- pro.rna.mean.cov %>% 
  mutate(xmin=mean.TPM,
         xmax=mean.TPM+RNA.CoV,
         ymin=mean.signal.length,
         ymax=mean.signal.length+protein.CoV
  )

pro.rna.mean.cov <- pro.rna.mean.cov %>%
  mutate(label=ifelse(Genes=="INS", "INS", ""))

cbPalette <- c("#E69F00", #lightorange
               "#56B4E9", #blue
               "#CC79A7", #magenta
               "#009E73", #green
               "#D55E00", #darkorange
               "#0072B2", #darkblue
               "#F0E442", #yellow
               "#999999" #grey
)

p <- ggplot(pro.rna.mean.cov,
       aes(
         y=mean.signal.length, 
         x=mean.TPM
         # colour= p_sig
       )) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4 , color="grey20") +
  
  geom_errorbar(aes(ymin = ymin,ymax = ymax) , #color=cbPalette[2],alpha=0.2
                color="grey30",alpha=0.3) + 
  geom_errorbarh(aes(xmin = xmin,xmax = xmax), #color=cbPalette[3],alpha=0.2
                 color="grey30",alpha=0.3) +
  geom_smooth(se=TRUE,
              method=lm
  ) +
  geom_text_repel(aes(label=label),size=4,
                  color="grey10",
                  box.padding   = 0.4,
                  point.padding = 0,
                  force=1,
                  force_pull=0,
                  hjust=-35,
                  max.overlaps = Inf, # always show all label, regardless of overlap
                 # min.segment.length = 0, # always draw line
                  segment.color = 'grey20')+
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
           )+
  
  labs(x= "RNA abundance (mean TPM)", 
       y= "Protein abundance (signal/length)"
  ) +
 # scale_colour_manual(values=c("red","blue","grey60")) +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                limits=c(0.1,150000),
                labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                limits=c(0.1,150000),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "bl") + 
  #  scale_x_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(0,120000)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

png(file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar'.png",
    width = 5, 
    height = 5, 
    units = "in", res = 600)

{pdf(#file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar_INS.pdf",
  file = "figures/RNA_protein_correlation_across_gene_INS_new.pdf",
    width = 5, 
    height = 5)

p
dev.off()}
```

# log2 has the same correlation as plot in log10 axis
```{r plot across gene correlation log2}
pro.rna.mean.cov <- read.csv("output/proteomics50_RNAseq50_anno_mean_CoV.csv")
pro.rna.mean.cov <- pro.rna.mean.cov %>% 
  mutate(xmin=log2(mean.TPM),
         xmax=log2(mean.TPM)+RNA.SD,
         ymin=log2(mean.signal.length),
         ymax=log2(mean.signal.length)+protein.length.SD
  )
# plot CoV as error bar (didn't use this)
pro.rna.mean.cov <- pro.rna.mean.cov %>% 
  mutate(xmin=mean.TPM,
         xmax=mean.TPM+RNA.CoV,
         ymin=mean.signal.length,
         ymax=mean.signal.length+protein.CoV
  )

pro.rna.mean.cov <- pro.rna.mean.cov %>%
  mutate(label=ifelse(Genes=="INS", "INS", ""))

cbPalette <- c("#E69F00", #lightorange
               "#56B4E9", #blue
               "#CC79A7", #magenta
               "#009E73", #green
               "#D55E00", #darkorange
               "#0072B2", #darkblue
               "#F0E442", #yellow
               "#999999" #grey
)

p <- ggplot(pro.rna.mean.cov,
       aes(
         y=log2(mean.signal.length), 
         x=log2(mean.TPM)
         # colour= p_sig
       )) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4 , color="grey20") +
  
#  geom_errorbar(aes(ymin = ymin,ymax = ymax) ,  color="grey30",alpha=0.3) + 
#  geom_errorbarh(aes(xmin = xmin,xmax = xmax), color="grey30",alpha=0.3) +
  geom_smooth(se=TRUE,
              method=lm
  ) +
  geom_text_repel(aes(label=label),size=4,
                  color="grey10",
                  box.padding   = 0.4,
                  point.padding = 0,
                  force=1,
                  force_pull=0,
                  hjust=-35,
                  max.overlaps = Inf, # always show all label, regardless of overlap
                 # min.segment.length = 0, # always draw line
                  segment.color = 'grey20')+
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
           )+
  
  labs(x= "RNA abundance (mean TPM)", 
       y= "Protein abundance (signal/length)"
  ) +
 # scale_colour_manual(values=c("red","blue","grey60")) +
  
  
#  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
#                limits=c(0.1,150000),
#                labels = trans_format("log10", math_format(10^.x))) +
#  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
#                limits=c(0.1,150000),
#                labels = trans_format("log10", math_format(10^.x))) +
#  annotation_logticks(sides = "bl") + 
  
  
  #  scale_x_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(0,120000)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

png(file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar'.png",
    width = 5, 
    height = 5, 
    units = "in", res = 600)

{pdf(#file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar_INS.pdf",
  file = "figures/RNA_protein_correlation_across_gene_INS_new.pdf",
    width = 5, 
    height = 5)

p
dev.off()}
```

```{r}
p <- ggplot(pro.rna.cor.all.gene.oneBatch,
            aes(
              y=mean.signal.length, 
              x=mean.TPM,
              colour= padj_sig
            )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(
    #aes(colour=R),
    #color="grey20",
    size=0.8,
    alpha= 0.4 
  ) +
#  geom_smooth(
    #se=TRUE,
#    se=FALSE,
#    method=lm
#             ) +
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
           )+
  
  labs(x= "RNA abundance (mean TPM)", 
       y= "Protein abundance (signal/length)"
  ) +
  scale_colour_manual(values=c("red","blue","grey60")) +
  #scale_fill_gradient2(high = "red3",mid = "white",low = "blue3",midpoint = 0,
  #  space = "Lab",na.value = "grey50",guide = "colourbar",aesthetics = "colour")+
  
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "bl") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position=
      "left",
      #"none",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = TRUE, groupFill = TRUE,
                 margins =c("both"), colour="#E69F00",fill=alpha("#E69F00",0.5)) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2

ggsave(filename="figures/RNA_protein_correlation_all_genes_TPM.png",width=5,height=5,units="in",dpi=600)

ggsave(filename="figures/RNA_protein_correlation_all_genes_TPM_trend_linear.png",width=5,height=5,units="in",dpi=600)
ggsave(filename="figures/RNA_protein_correlation_all_genes_TPM_trend_curve_oneBatch.png",width=5,height=5,units="in",dpi=600)

png(file = "figures/RNA_protein_correlation_all_gene_padj_TPM_color_density_oneBatch.png",
    width = 9, 
    height = 6.5, 
    units = "in", res = 600)
png(file = "figures/RNA_protein_correlation_all_gene_padj_TPM_color_density_linear_oneBatch.png",
    width = 9, 
    height = 6.5, 
    units = "in", res = 600)

pdf(file = "figures/RNA_protein_correlation_all_gene_padj_TPM_color_density_oneBatch.pdf",
    width = 9, 
    height = 6.5)

p2
dev.off()


# Are the correlations different?
cocor.indep.groups(r1.jk=0.54, r2.hm=0.48, n1=1412, n2=5475, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
cocor.indep.groups(r1.jk=0.54, r2.hm=0.47, n1=1412, n2=98, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
cocor.indep.groups(r1.jk=0.48, r2.hm=0.47, n1=5475, n2=98, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
# Diedenhofen, B. & Musch, J. (2015). cocor: A Comprehensive Solution for the Statistical Comparison of Correlations. PLoS ONE, 10(4): e0121945. doi:10.1371/journal.pone.0121945


# Plot variance for oneBatch ==================

# protein and RNA CoV ===================

colnames(pro.rna.cor.all.gene.oneBatch)
common.genes <- pro.rna.cor.all.gene.oneBatch$ENSG
length(common.genes) # 6918 genes
common.individuals <- rownames(rna.cor2) 
length(common.individuals) # 90
View(tpms.oneBatch)
colnames(tpms.oneBatch)
tpms.cv.oneBatch <- tpms.oneBatch

tpms.cv.oneBatch$RNA.TPM.CoV <- apply(tpms.oneBatch[,common.individuals], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
tpms.cv.oneBatch$RNA.TPM.SD <- apply(tpms.oneBatch[,common.individuals], 1, function(x) sd(x, na.rm=T))

#tpms.cv$rank <- nrow(tpms.5):1
tpms.cv.oneBatch <- tpms.cv.oneBatch %>% 
  relocate(RNA.TPM.CoV:RNA.TPM.SD, .after = mean.TPM)
View(tpms.cv.oneBatch)
colnames(tpms.cv.oneBatch)

# calculate CoV for proteins (normalized to length)
View(pro.islet.filter.norm.oneBatch)
colnames(pro.islet.filter.norm.oneBatch)
which(pro.islet.filter.norm.oneBatch$Genes.x=="INS")
pro.islet.filter.norm.oneBatch[2248,"Length"] <- 51
pro.islet.filter.norm.oneBatch$mean.signal.length <- pro.islet.filter.norm.oneBatch$mean.signal/pro.islet.filter.norm.oneBatch$Length
pro.islet.norm.cv.oneBatch <- pro.islet.filter.norm.oneBatch %>% 
  mutate_at(vars(10:99),list(norm.length = ~./Length)) # every sample & gene is divided by length
#pro.islet.norm.cv.oneBatch <- pro.islet.filter.norm.oneBatch %>% 
#  mutate_at(vars(12:101),list(norm.length = ~./Length)) # every sample & gene is divided by length
dim(pro.islet.filter.norm.oneBatch)
dim(pro.islet.norm.cv.oneBatch)
colnames(pro.islet.norm.cv.oneBatch)
View(pro.islet.norm.cv.oneBatch[pro.islet.norm.cv.oneBatch$Genes.x=="INS",])
common.individuals.norm <- paste0(common.individuals,"_norm.length")

pro.islet.norm.cv.oneBatch$protein.CoV <- apply(pro.islet.norm.cv.oneBatch[,common.individuals.norm], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
pro.islet.norm.cv.oneBatch$protein.SD <- apply(pro.islet.norm.cv.oneBatch[,common.individuals.norm], 1, function(x) sd(x, na.rm=T))
#pro.islet.norm.cv.oneBatch$protein.CoV <- apply(pro.islet.norm.cv.oneBatch[,149:235], 1, function(x) sd(x, na.rm=T) / mean(x, na.rm=T) * 100)
#pro.islet.norm.cv.oneBatch$protein.SD <- apply(pro.islet.norm.cv.oneBatch[,149:235], 1, function(x) sd(x, na.rm=T))

pro.islet.norm.cv.oneBatch <- pro.islet.norm.cv.oneBatch %>% 
  relocate(protein.CoV:protein.SD, .after = mean.signal.length)
View(pro.islet.norm.cv.oneBatch)
colnames(pro.islet.norm.cv.oneBatch)

pro.rna.cv.all.gene.oneBatch <- pro.islet.norm.cv.oneBatch[,1:11] %>%
  left_join(tpms.cv.oneBatch[,c(1:11)], by=c("ENSG"="ensembl_gene_id"))
#pro.rna.cv.all.gene.oneBatch <- pro.islet.norm.cv.oneBatch[,1:13] %>%
#  left_join(tpms.cv.oneBatch[,c(1:11)], by=c("ENSG"="ensembl_gene_id"))
colnames(pro.rna.cv.all.gene.oneBatch)
dim(pro.rna.cv.all.gene.oneBatch) # 6918x21, SOD2 still missing due to outdated ENSG

pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>%
  left_join(pro.rna.cor.each.gene.oneBatch[,1:6], by="ENSG") # added the within-gene across-sample correlations

View(pro.rna.cv.all.gene.oneBatch)
pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch[order(pro.rna.cv.all.gene.oneBatch$mean.signal,decreasing = T),]
pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch[!duplicated(pro.rna.cv.all.gene.oneBatch$Protein.Ids),]

## CoV XY dot plot --------------
p <- ggplot(pro.rna.cv.all.gene.oneBatch,
            aes(
              y=protein.CoV, 
              x=RNA.TPM.CoV,
              
              colour= padj_sig
            )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4   #, color="grey20"
  ) +
  
#    geom_smooth(
#      se=FALSE,
#      method=lm
#    )+
  
    stat_cor(method = "pearson" #, label.x = 3, label.y = 30
   )+
  
  labs(x= "RNA (TPM) Coefficient of Variation", 
       y= "Protein Coefficient of Variation"
  ) +
  scale_colour_manual(values=c("red","blue","grey60")) +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "bl") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    
    legend.position="left",
    
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

p2 <- ggMarginal(p, type = "density", groupColour = TRUE, groupFill = TRUE,
                 margins =c("both"), colour="#E69F00",fill=alpha("#E69F00",0.5)) #https://jtr13.github.io/cc21fall2/tutorial-for-scatter-plot-with-marginal-distribution.html
p2

ggsave(filename="figures/RNA_protein_CoV_oneBatch.png",width=5,height=5,units="in",dpi=600)
ggsave(filename="figures/RNA_protein_CoV_curve_oneBatch.png",width=5,height=5,units="in",dpi=600)
ggsave(filename="figures/RNA_protein_CoV_linear_oneBatch.png",width=5,height=5,units="in",dpi=600)


png(file = "figures/RNA_protein_CoV_color_R_oneBatch.png", 
    width = 8, 
    height = 6.5, 
    units = "in", res = 600)

png(file = "figures/RNA_protein_CoV_color_R_linear_oneBatch.png", 
    width = 8, 
    height = 6.5, 
    units = "in", res = 600)

p2
dev.off()

# Are the correlations different?
cocor.indep.groups(r1.jk=0.62, r2.hm=0.25, n1=1412, n2=5475, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
cocor.indep.groups(r1.jk=0.62, r2.hm=0.39, n1=1412, n2=98, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
cocor.indep.groups(r1.jk=0.25, r2.hm=0.39, n1=5475, n2=98, alternative="two.sided", 
                   alpha=0.05, conf.level=0.95, null.value=0)
# Diedenhofen, B. & Musch, J. (2015). cocor: A Comprehensive Solution for the Statistical Comparison of Correlations. PLoS ONE, 10(4): e0121945. doi:10.1371/journal.pone.0121945




# rank by TPM or protein (norm)
colnames(pro.rna.cv.all.gene.oneBatch)

pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>% 
  arrange(mean.signal.length) 
View(pro.rna.cv.all.gene.oneBatch)
pro.rna.cv.all.gene.oneBatch$rank.protein <- 1:nrow(pro.rna.cv.all.gene.oneBatch)

pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>% 
  arrange(mean.TPM) 
View(pro.rna.cv.all.gene.oneBatch)
pro.rna.cv.all.gene.oneBatch$rank.TPM <- 1:nrow(pro.rna.cv.all.gene.oneBatch)

pro.rna.cv.all.gene.long.oneBatch <- pro.rna.cv.all.gene.oneBatch %>%
  pivot_longer(cols = c(RNA.TPM.CoV,protein.CoV), names_to = "RNA.or.protein", values_to = "CoV")
View(pro.rna.cv.all.gene.long.oneBatch)

## protein and RNA CoV ~ TPM rank plot ---------------

p <- ggplot(pro.rna.cv.all.gene.long.oneBatch,
            aes(
              #y=TPM.CoV, 
              y=CoV,
              x=rank.TPM,
              
              colour= RNA.or.protein
            )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.2 #, color="grey20"
  ) +
  geom_smooth(
    se=F,
    aes(group=RNA.or.protein),
    #colour=c("grey40","blue"),
    #              #method=lm
  )+
  #stat_cor(method = "pearson" #, label.x = 3, label.y = 30
  #)+
  
  labs(x= "Rank of RNA abundance (TPM)", 
       #y= "RNA (TPM) Coefficient of Variation"
       y= "Coefficient of Variation"
  ) +
  
  #scale_colour_manual(values=c("red","blue","grey60")) +
  scale_colour_manual(values=cbPalette[2:3]) +
  
  # scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
  #                labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") + 
  #  scale_x_continuous(breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(1,9.5)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    
    legend.position="top",
    #legend.position="none",
    
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

ggsave(filename="figures/RNA_protein_CoV_rankTPM_trend_curve_oneBatch.png",width=5,height=6.5,units="in",dpi=600)
ggsave(filename="figures/RNA_protein_CoV_rankTPM_trend_curve_oneBatch.pdf",width=5,height=6.5,units="in")


ggsave(filename="figures/RNA_protein_CoV_rankTPM_trend_curve_shade_oneBatch.png",width=5,height=6.5,units="in",dpi=600)


# error bar

#ggplot(data = df,aes(x = x,y = y)) + 
#  geom_point() + 
#  geom_errorbar(aes(ymin = ymin,ymax = ymax)) + 
#  geom_errorbarh(aes(xmin = xmin,xmax = xmax))

# plot SD as error bar (did this)
pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>% 
  mutate(xmin=mean.TPM,
         xmax=mean.TPM+RNA.TPM.SD,
         ymin=mean.signal.length,
         ymax=mean.signal.length+protein.SD
  )
# plot CoV as error bar (didn't use this)
pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>% 
  mutate(xmin=mean.TPM,
         xmax=mean.TPM+RNA.TPM.CoV,
         ymin=mean.signal.length,
         ymax=mean.signal.length+protein.CoV
  )

colnames(pro.rna.cv.all.gene)
View(pro.rna.cv.all.gene.oneBatch)
pro.rna.cv.all.gene.oneBatch <- pro.rna.cv.all.gene.oneBatch %>%
  mutate(label=ifelse(Genes.x=="INS", "INS", ""))

cbPalette <- c("#E69F00", #lightorange
               "#56B4E9", #blue
               "#CC79A7", #magenta
               "#009E73", #green
               "#D55E00", #darkorange
               "#0072B2", #darkblue
               "#F0E442", #yellow
               "#999999" #grey
)

p <- ggplot(pro.rna.cv.all.gene.oneBatch,
       aes(
         y=mean.signal.length, 
         x=mean.TPM,
         
         # colour= p_sig
       )
) +  
  #geom_vline(xintercept = 0,linetype="dashed", color = "grey50")+
  #geom_hline(yintercept = 0,linetype="dashed", color = "grey50")+
  
  geom_point(size=0.8, alpha= 0.4 , color="grey20"
  ) +
  
  geom_errorbar(aes(ymin = ymin,ymax = ymax) , #color=cbPalette[2],alpha=0.2
                color="grey30",alpha=0.3) + 
  geom_errorbarh(aes(xmin = xmin,xmax = xmax), #color=cbPalette[3],alpha=0.2
                 color="grey30",alpha=0.3) +
  geom_smooth(se=TRUE,
              method=lm
  ) +
  geom_text_repel(aes(label=label),size=4,
                  color="grey10",
                  box.padding   = 0.4,
                  point.padding = 0,
                  force=1,
                  force_pull=0,
                  hjust=-35,
                  max.overlaps = Inf, # always show all label, regardless of overlap
                 # min.segment.length = 0, # always draw line
                  segment.color = 'grey20')+
  stat_cor(method = "pearson" #, label.x = 3, label.y = 30
           )+
  
  labs(x= "RNA abundance (mean TPM)", 
       y= "Protein abundance (signal/length)"
  ) +
  scale_colour_manual(values=c("red","blue","grey60")) +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                limits=c(0.1,150000),
                labels = trans_format("log10", math_format(10^.x))) +
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x),
                limits=c(0.1,150000),
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "bl") + 
  #  scale_x_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #                     limits= c(0,120000)) +
  #  scale_y_continuous(#breaks=c(0,2,4,6,8,10,12,14),
  #    limits= c(0,20)) +
  #scale_x_continuous(limits = ~ c(-1, 1) * max(abs(.x))) +
  #scale_y_continuous(limits = ~ c(-1, 1) * max(abs(.x)))+
  
  theme_classic() +
  theme(
    axis.title.y = element_text(colour = "black",size=14),
    axis.title.x = element_text(colour = "black",size=14),
    axis.text.y = element_text(colour = "black",size=12),
    axis.text.x = element_text(colour = "black",size=12),
    #axis.ticks.x = element_blank(),
    #legend.title = element_blank(),
    legend.position="right",
    legend.title=element_text(size=14), 
    legend.text=element_text(size=14),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    aspect.ratio=1/1) +
  
  labs(color = "Correlation" #expression(atop("Differentially",paste("expressed")))
  )+
  theme(axis.line = element_line()) +
  guides(colour = guide_legend(override.aes = list(size=2,alpha=1)))
p

png(file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar'.png",
    width = 5, 
    height = 5, 
    units = "in", res = 600)

pdf(file = "figures/RNA_protein_correlation_all_gene_error_SD_oneBatch_greybar_INS.pdf",
    width = 5, 
    height = 5)

p
dev.off()


```
